src/Events/Calendar_Embeds/Render.php

```diff
@@ -0,0 +1,1306 @@
+ *
+ * @since TBD
+ *
+ * @package TEC/Events/Calendar_Embeds
+ */
+class Render {
+	/**
+	 * Prefix for the transient where we will save the base values for the
+	 * setup of the context of the embed.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	const TRANSIENT_PREFIX = 'tec_events_calendar_embeds_view_params_';
+
+	/**
+	 * Arguments of the current view.
+	 *
+	 * @since TBD
+	 *
+	 * @var array
+	 */
+	protected $arguments;
+
+	/**
+	 * {@inheritDoc}
+	 *
+	 * @since TBD
+	 *
+	 * @var array
+	 */
+	protected $default_arguments = [
+		'author'               => null,
+		'category'             => null,
+		'container-classes'    => [],
+		'date'                 => null,
+		'events_per_page'      => null,
+		'exclude-category'     => null,
+		'exclude-tag'          => null,
+		'featured'             => null,
+		'filter-bar'           => false,
+		'hide_weekends'        => false,
+		'hide-datepicker'      => false,
+		'hide-export'          => false,
+		'id'                   => null,
+		'is-widget'            => false,
+		'jsonld'               => true,
+		'keyword'              => null,
+		'layout'               => 'vertical', // @todo Change to auto when we enable that option.
+		'main-calendar'        => false,
+		'month_events_per_day' => null,
+		'organizer'            => null,
+		'past'                 => false,
+		'should_manage_url'    => false, // @todo @bordoni @lucatume @be Update this when URL management is fixed.
+		'skip-empty'           => false,
+		'tag'                  => null,
+		'tax-operand'          => 'AND',
+		'tribe-bar'            => false,
+		'venue'                => null,
+		'view'                 => null,
+		'week_events_per_day'  => null,
+		'week_offset'          => null,
+	];
+
+	/**
+	 * Setup the arguments for the view.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $arguments Arguments to be used to setup the view.
+	 */
+	public function setup( array $arguments ): void {
+		$this->arguments = wp_parse_args( $arguments, $this->default_arguments );
+	}
+
+	/**
+	 * Toggles the filtering of URLs to match the place where this is called.
+	 *
+	 * @since TBD
+	 *
+	 * @param bool $toggle Whether to turn the hooks on or off.
+	 *
+	 * @return void
+	 */
+	protected function toggle_view_hooks( $toggle ): void {
+		if ( $toggle ) {
+			$this->add_view_hooks();
+		} else {
+			$this->remove_view_hooks();
+		}
+
+		/**
+		 * Fires after View hooks have been toggled while rendering.
+		 *
+		 * @since TBD
+		 *
+		 * @param bool   $toggle Whether the hooks should be turned on or off.
+		 * @param static $this   The instance that is toggling the View hooks.
+		 */
+		do_action( 'tec_events_calendar_embeds_render_toggle_view_hooks', $toggle, $this );
+	}
+
+	/**
+	 * Toggles on portions of the template based on the params.
+	 *
+	 * @since TBD
+	 */
+	protected function add_view_hooks(): void {
+		add_filter( 'tribe_events_views_v2_url_query_args', [ $this, 'filter_view_query_args' ], 15, 3 );
```

Comment:
```suggestion
		add_filter( 'tribe_events_views_v2_url_query_args', [ $this, 'filter_view_query_args' ], 15, 1 );
```
Or - change the callback to take all three, but then of course you'll get tagged for unused params 😆